google/cloud/sql/connector/connector.py

```diff
@@ -130,10 +130,22 @@ def __init__(
         # set default params for connections
         self._timeout = timeout
         self._enable_iam_auth = enable_iam_auth
-        self._ip_type = ip_type
         self._quota_project = quota_project
         self._sqladmin_api_endpoint = sqladmin_api_endpoint
         self._user_agent = user_agent
+        # if ip_type is str, convert to enum
+        if isinstance(ip_type, str):
+            if ip_type.lower() == "public":
+                ip_type = IPTypes.PUBLIC
+            elif ip_type.lower() == "private":
+                ip_type = IPTypes.PRIVATE
+            elif ip_type.lower() == "psc":
+                ip_type = IPTypes.PSC
+            else:
+                raise ValueError(
+                    f"Incorrect value for ip_type, got '{ip_type}'. Want one of: 'public', 'private' or 'psc'."
+                )
+        self._ip_type = ip_type
```

Comment:
Technically this code could be removed and we could allow the conversion to purely happen once during the `connect` call. 

However, this would cause the case where users configure the ip type in the Connector() to be converted repeatedly on every single call to connect. For this reason I thought it may be worth converting in the Connector.__init__ so this case is converted once and then skips the check in the `connect` for reduced latency on connect.